### ISSUE TITLE
Error occurs reporting another error - Anonymous class should declare a `graphql_name` 

### ISSUE THREAD

**Issue by Aryk** [rmosolgo/graphql-ruby#3203]
I spent a few hours debugging the wrong issue...this actually happened to this other user: 

https://github.com/rmosolgo/graphql-ruby/issues/2385

I was getting this strange error which is actually not applicable to my code...there is no Type that doesn't have a graphql_name...after some digging I found out that I was getting an error in an error catch:

https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/field.rb#L506

My suggestion...change code so that it doesn't throw out an error which catching another one:

```ruby
rescue StandardError => err
        raise ArgumentError, "Failed to build return type for #{@owner&.graphql_name || "(Unknown)"}.#{name} from #{@return_type_expr.inspect}: (#{err.class}) #{err.message}", err.backtrace
      end
```

Current version: 1.11.3

**Comment by rmosolgo** [comment]
Hey, thanks for reporting this. I'd like to improve the code, but I don't quite understand what went wrong. Could you share the full error and backtrace that you encountered (even though it was the "wrong" error)?

**Comment by Aryk** [comment]
I can't repro it because I've changed the code a ton since yesterday ;)

What I can say was that the culprit was that I tried to call a class that doesn't exist:

```ruby
module Types
  class SomeType < Types::BaseSomeType[MispelledClassNam]
  end
end
```

The `Types::BaseSomeType` returns a     Class.new(Types::BaseObject) instance but obviously couldnt complete because the error came much earlier.

**Comment by rmosolgo** [comment]
> I added "&" to your code

That's what I don't quite understand. Doesn't `&` skip the method call _if the receiver is `nil`_? Does that mean `@owner` was nil in that case? 

If that was true, I would have expected a `NoMethodError`, not the "anonymous class should declare a GraphQL name" error 🤔

**Comment by Aryk** [comment]
That's right...let me look into it some more...but it was something on that line...I must have forgotten something. Need to stop coding at 4am. 😄
